application/src/main/java/org/togetherjava/tjbot/commands/system/BotCore.java

```diff
@@ -309,25 +317,25 @@ public void onUserContextInteraction(final UserContextInteractionEvent event) {
     }
 
     /**
-     * Returns the {@link ComponentId} of the given componentId. If the component has expired, the
-     * user gets a message stating that.
+     * Returns the {@link ComponentId} instace of the given componentId. If the component has
+     * expired, the user gets a message stating that.
      *
      * @param event the {@link IReplyCallback event} to reply to
      * @param componentId the component's ID
-     *
      * @return the relating {@link ComponentId}
      */
     @Nullable
-    public ComponentId handleGetComponentId(IReplyCallback event, String componentId) {
+    public Optional<ComponentId> handleParseComponentId(IReplyCallback event, String componentId) {
         Optional<ComponentId> componentIdOpt;
         try {
             componentIdOpt = componentIdParser.parse(componentId);
         } catch (InvalidComponentIdFormatException e) {
             logger
-                .error("Unable to route event (#{}) back to its corresponding user interactor. The component ID was in an unexpected format."
+                .error("Unable to route event (#{}) back to its corresponding {}}. The component ID was in an unexpected format."
                         + " All button and menu events have to use a component ID created in a specific format"
-                        + " (refer to the documentation of UserInteractor). Component ID was: {}",
-                        event.getId(), componentId, e);
+                        + " (refer to the documentation of {}}). Component ID was: {}",
+                        event.getId(), UserInteractor.class.getSimpleName(),
+                        UserInteractor.class.getSimpleName(), componentId, e);
```

Comment:
there is actually specific syntax for reusing an argument instead of specifying it multiple times. i think its like `{1}` or something.

that feature also exists for javas formatter, syntax is like `%1$s` instead of just `%s`

Reply:
do you have a link on this? I can't find it at all, nor does `{1}` seem to work

Reply:
https://docs.oracle.com/javase/1.5.0/docs/api/java/util/Formatter.html#syntax @Tais993 
do `%2$s` to use the second value

Reply:
This doesn't work, we're using SLF4J, not Java's Formatter

Reply:
ohh yea, my bad

Reply:
I got it from this: https://stackoverflow.com/a/15881018/2411243

Maybe u have to use an array for it to work

Reply:
Unfortunately, I couldnt find anything in the docs: https://www.slf4j.org/api/org/slf4j/helpers/MessageFormatter.html

Reply:
try around and if u cant get it to work, feel free to resolve this conversation. thanks

Reply:
any luck with that? otherwise resolve it and ping me. ill accept then.

Reply:
@Tais993 ?

Reply:
Mb, I tried but couldn't get it to work. Not sure or an array works, but not gonna try that (dont find it any better personally)

Reply:
fair